google/cloud/firestore_v1/base_query.py

```diff
@@ -756,8 +764,9 @@ def end_before(
         )
 
     def end_at(
-        self, document_fields_or_snapshot: Union[DocumentSnapshot, dict, list, tuple]
-    ) -> "BaseQuery":
+        self: QueryType,
```

Comment:
nit: it shouldn't be necessary to include type annotations for self (mypy doesn't require it, and we typically don't include them in other GCP libraries). 

Is there a reason it's needed here?